packages/sommelier.rb

```diff
@@ -3,109 +3,60 @@
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> initsommelier"
-      system "echo 'if [ -z \"\$SOMM\" ]; then' >> initsommelier"
-      system "echo '  [ -f #{CREW_PREFIX}/bin/stopbroadway ] && stopbroadway' >> initsommelier"
-      system "echo '  #{CREW_PREFIX}/sbin/sommelierd &' >> initsommelier"
-      system "echo '  sleep 3' >> initsommelier"
-      system "echo 'fi' >> initsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> initsommelier"
-      system "echo 'if [ ! -z \"\$SOMM\" ]; then' >> initsommelier"
-      system "echo '  echo \"sommelier process \$SOMM is running\"' >> initsommelier"
-      system "echo 'else' >> initsommelier"
-      system "echo '  echo \"sommelier failed to start\"' >> initsommelier"
-      system "echo '  exit 1' >> initsommelier"
-      system "echo 'fi' >> initsommelier"
-      system "echo '#!/bin/bash' > stopsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> stopsommelier"
-      system "echo 'if [ ! -z \"\$SOMM\" ]; then' >> stopsommelier"
-      system "echo '  killall -g sommelier' >> stopsommelier"
-      system "echo '  sleep 3' >> stopsommelier"
-      system "echo 'fi' >> stopsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> stopsommelier"
-      system "echo 'if [ -z \"\$SOMM\" ]; then' >> stopsommelier"
-      system "echo '  echo \"sommelier process stopped\"' >> stopsommelier"
-      system "echo 'else' >> stopsommelier"
-      system "echo '  echo \"sommelier process \$SOMM is running\"' >> stopsommelier"
-      system "echo '  exit 1' >> stopsommelier"
-      system "echo 'fi' >> stopsommelier"
+    system 'git', 'clone', 'https://chromium.googlesource.com/chromiumos/containers/sommelier'
+    Dir.chdir ('sommelier') do
+      system 'git', 'reset', '--hard', version
+      system 'sed', '-i', "s,/lib/,/#{ARCH_LIB}/,g", 'Makefile'
+      system 'sed', '-i', "s,-I.,& -I#{CREW_PREFIX}/include/pixman-1,g", 'Makefile'
+      system 'make', "PREFIX=#{CREW_PREFIX}", "SYSCONFDIR=#{CREW_PREFIX}/etc"
     end
   end
 
   def self.install
-    Dir.chdir ("sommelier") do
-      system "make", "PREFIX=#{CREW_PREFIX}", "SYSCONFDIR=#{CREW_PREFIX}/etc", "DESTDIR=#{CREW_DEST_DIR}", "install"
-      system "install -Dm755 sommelierd #{CREW_DEST_PREFIX}/sbin/sommelierd"
-      system "install -Dm755 initsommelier #{CREW_DEST_PREFIX}/bin/initsommelier"
-      system "install -Dm755 stopsommelier #{CREW_DEST_PREFIX}/bin/stopsommelier"
-      system "install -Dm644 .sommelier.env #{CREW_DEST_HOME}/.sommelier.env"
+    Dir.chdir ('sommelier') do
+      system 'make', "PREFIX=#{CREW_PREFIX}", "SYSCONFDIR=#{CREW_PREFIX}/etc", "DESTDIR=#{CREW_DEST_DIR}", 'install'
     end
   end
 
   def self.postinstall
+    puts 'To configure the Sommelier daemon, remove the former'.lightblue
+    puts ' "Sommelier environment variables + daemon" section'.lightblue
+    puts ' from `~/.bashrc`, and run this:'.lightblue
```

Comment:
Dito here.  You made the transition more difficult.

Reply:
See above. I wish there was a better solution to running X apps then this old version of Sommelier.

Reply:
I have an idea: Create a `virtwl` package with the required header and build from there.

Reply:
Argh. It doesn't work.